src/components/ProjectPreview/index.tsx

```diff
@@ -12,38 +14,66 @@ export const ProjectPreview: FC<PublicProject> = ({
 }) => {
   const parentRef = useRef<HTMLDivElement>(null);
   const [svgWrapperWidth, setSvgWrapperWidth] = useState(0);
-
-  const updateWidthAndHeight = (): void => {
-    if (parentRef.current === null) return;
-    setSvgWrapperWidth(parentRef.current.offsetWidth);
-  };
+  const [svgWrapperHeight, setSvgWrapperHeight] = useState(0);
+  const [isInViewport, mapWrapperRef] = useIsInViewport({ threshold: 50 });
 
   useEffect(() => {
-    if (parentRef.current === null) return;
-    setSvgWrapperWidth(parentRef.current.offsetWidth);
+    const updateWidthAndHeight = (): void => {
+      if (parentRef.current === null) return;
+      setSvgWrapperWidth(parentRef.current.offsetWidth);
+      setSvgWrapperHeight(parentRef.current.offsetHeight);
+    };
 
     window.addEventListener("resize", updateWidthAndHeight);
+    updateWidthAndHeight();
 
     return () => window.removeEventListener("resize", updateWidthAndHeight);
-  }, []);
+  }, [parentRef]);
 
   return (
     <div
+      ref={parentRef}
       className={[
         "bg-white shadow-lg hover:bg-blue-25",
         "border border-blue-50",
         "cursor-pointer transition rounded-md",
-        "relative overflow-hidden",
+        "relative overflow-hidden group",
       ].join(" ")}
       style={{ paddingBottom: 100 }}
     >
-      <Link href={`${id}`}>
-        <a sx={{ textDecoration: "none", color: "text" }}>
+      <Link href={`/${id}`}>
+        <a href={`/${id}`} ref={mapWrapperRef}>
+          {isInViewport && (
```

Comment:
Is there any way that we can cache what has already been in the viewport? Right now the map and the Mapbox tiles are fetched every time something comes **back** into the viewport.

Or is this on purpose because rendering so much WebGL at the same time is expensive?

Reply:
Thanks @dnsos !

Only maps visible in the viewport are rendered and the tiles are actually loaded from the cache. I didn't notice too many problems performance-wise but there are certainly some other improvements. I'm not sure about limiting the number of projects though as it could result in a page with only some of the projects having a background map, which wouldn't look good I think. 

About the maps coming back, I agree. When we use reactQuery we can say to fetch only once and then never again. I didn't achieve this using useSWR. However, I think the results are fetched from the cache anyway but we might want to investigate further. 

Maybe we should merge this, so people can test it and then create a new ticket if performance problems persist? What do you think?

Reply:
> is this on purpose because rendering so much WebGL at the same time is expensive?

Exactly. Too many canvas on the DOM is bad for performance...

Reply:
Thanks for the explanation. And I absolutely agree, let's merge this and observe the performance and how people perceive the maps! 🙂